Modules/HR/Http/Controllers/EmployeeController.php

```diff
@@ -30,8 +31,33 @@ public function index(Request $request)
     {
         $filters = $request->all();
         $filters = $filters ?: $this->service->defaultFilters();
+        $name = request('name');
+        $users = Role::where('name', $name)->first()->users()->get();
+        $data = [];
+        foreach ($users as $user) {
+            $data[] = $user->id;
+        }
+        $employees = Employee::filterbyrole($data)->get();
 
-        return view('hr.employees.index', $this->service->index($filters));
+        return view('hr.employees.index', $this->service->index($filters))->with([
+            'employees' => $employees,
+        ]);
+    }
+
+    public function filterEmployee(Request $request, $name)
+    {
+        $filters = $request->all();
+        $filters = $filters ?: $this->service->defaultFilters();
+        $users = Role::where('name', $name)->first()->users()->get();
```

Comment:
Can you try with this, and see if it works. I think this will give you all the users that are employees, and you can iterate on the users instead of employees in the blade file 
```suggestion
        $users = Role::where('name', $name)->first()->users()->whereHas('employees')->get();
```
`employees` can be a relationship between user and employee